plugins/woocommerce-admin/client/products/edit-product-page.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { TextControl } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { ProductFormLayout } from './layout/product-form-layout';
+
+const EditProductPage: React.FC = () => {
+	return (
+		<div>
+			<ProductFormLayout
+				categories={ [
+					{
+						id: 'product_details',
+						title: 'Product details',
+						description:
+							'This info will be displayed on the product page, category pages, social media, and search results.',
+						fields: [
+							<TextControl
```

Comment:
I am not a 100% set on passing fields in like this, of-course these fields could be defined in a separate file and exported.

My other potential thought was to only pass in the field definition, like `name`, `label`, and `type` and render that within. I think this could be overkill though given a large amount of fields are unique it seems, the only overlapping one is the text field.

Instead of passing in a list of `fields` we could also just pass in one `CategorySection` component that is unique for each section. This doesn't feel very re-usable though.

Reply:
Some questions about extensibility that I don't have answered to. I think they may help shape how we define things though.

- Can an extension add a field in-between existing fields? I think that would be useful, though easily abused.

- Can an extension hide or override an existing field? Again, useful but easily abused.

Reply:
Feels kind of odd to pass the categories in through a prop instead of as child components. But, I guess passing as a prop would make it easier to define the categories externally and pull them in.

Reply:
Some example code for the different options would help to visualize what the dev experience would be.